### DISCUSSION TITLE
Remote participation

### DISCUSSION THREAD

**Discussion by The-Compiler** [pytest-dev/sprint#8] [General]
Some people can't make it to the sprint, but might possibly be interested in remote participation. From the top of my head: @WarrenTheRabbit @jaraco and perhaps @okken @Zac-HD @nicoddemus @asottile (no pressure if this doesn't change anything for you, I just thought I'd let people know who can't make it to the sprint but might have wanted to participate).

This hasn't really been a priority for me to organize thus far, as my resources are somewhat scarce currently, and getting on-site participants finalized is the more time-critical aspect. However, Omicron (where we'll host the sprint) has WebEx meeting systems in all their meeting rooms, so it *should* be possible for us to get a couple of people in remotely as well.

Obviously it gets more difficult to coordinate things and also especially to mentor people in a hybrid format, but I think it'd be great as an opportunity for people to join who already know their way around pytest (at least on a basic level). I'll post more about the topic in this thread as soon as the on-site stuff is taken care of.

**Comment by nitzmahone** [comment]
I might be interested in participating for a few hours here and there remotely from UTC-8 - there are several proposed enhancements I'd be happy to pitch in on for design/impl (proper ExceptionGroup support, fixture control sans-name-magic, argparse usage slicing) and I've recently hit a few things in adjacent areas where I've had to roll my own something that could be useful to others in-box. No worries if there isn't a critical mass for "proper" remote setup too, though- it's tough to get the same collaborative experience with only 1-2 folks remote...

**Comment by The-Compiler** [comment]
# Remote topics and availability

After thinking about how to best do this and discussing this with the venue, we got to the conclusion that it'll be a bit tricky to also get remote participants in - given we have about an 1:1 to 1:2 ratio of core maintainers and new(-ish) contributors on-site, it will probably be tricky to have remote participants integrated nicely as well. We also decided we don't want the video conference system to be running for the whole sprint, because it's a bit of a weird feeling to be live-streamed all day :upside_down_face: 

Thus, my proposal would be that everyone who is interested in remote participation leaves a post in this discussion, with:

- Topics you are interested in
- Your timezone and availability (days/time). The sprint will be running Mon-Fri, [approximately 9:30-17:30 CEST](https://www.timeanddate.com/worldclock/fixedtime.html?msg=pytest+sprint&iso=20240617T0930&p1=2719&ah=8) most days (we might end a bit earlier on days where we have some events planned; or later if things happen to go very well!)
- A good way to reach you, ideally via Discord or IRC

I'll then try to summarize and announce those topics on-site when the sprint starts on Monday. If someone on-site is interested in working on one of those topics, they will reach out to you, and then we can always use the meeting room system thingy for a (hopefully) rather convenient video call into the room.

We have a `#sprint` channel in the [pytest Discord](https://discord.com/invite/pytest-dev) which is what I'd recommend for communication around the sprint.

If you want to work on your own thing, you're also most welcome to do that! I'll try to keep an eye on Discord, IRC, and incoming PRs during the week. Outside of that, I'm afraid I'll most likely not have the capacity to e.g. do longer 1:1 sessions with remote people, as I'll be quite busy with what's happening on-site.

Admittedly this is all a bit experimental! I really don't know how well this will work, and as @nitzmahone said above, it's tricky to get hybrid settings right (even more so since we'll probably be 2-3 groups on-site). I suppose we'll need to try and see how it goes.

---

@nitzmahone could you edit your post to add availability details there too?

@warrentherabbit @niloth-p @jaraco can you perhaps add such a post in this discussion too?

**Comment by Niloth-p** [comment]
Thank you for the update!

- Topics: I haven't picked any specific topics to work on yet. As someone pretty new to the repo, I'll have to identify easy areas I can pick up and contribute fast. If anyone has any suggestions, I'd love to hear them.
- Availability: I'm available throughout the week. I might have to take some short breaks in-between, but I can make it for most hours each day during the sprint hours, primarily 9:30 - 14:30 CEST.
- Discord tag: niloth_p_92383

**Comment by jaraco** [comment]
Thanks for reaching out. 

- Topics: I'm primarily interested in having more discussions about https://github.com/pytest-dev/pytest/discussions/12433. I'd be delighted to present on the Coherent System and answer questions to help give a richer understanding of the motivations. I'm also willing to help work on refactoring `pathlib` to facilitate supported hooks or functionality to enable the essential layout.
- I'm in US Eastern, so available in the latter half of each day (primarily 15:00-18:00 CEST). Monday and Tuesday I have intermittent availability, so would prefer to book dedicated time on Wed-Fri.
- I'm on Discord as jaraco and have joined the pytest-dev server.

**Reply by RonnyPfannschmidt** [reply]
The coherent system needs some more explanation, at first glance it triggers a gut wrench 

It may fit the import roots system I envision, but my initial idea on it's edge cases is giving me hives
